eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Bbody.hs

```diff
@@ -142,27 +133,81 @@ instance
   ) =>
   EncCBOR (ConwayBbodyPredFailure era)
   where
-  encCBOR =
-    encode . \case
-      WrongBlockBodySizeBBODY x y -> Sum WrongBlockBodySizeBBODY 0 !> To x !> To y
-      InvalidBodyHashBBODY x y -> Sum (InvalidBodyHashBBODY @era) 1 !> To x !> To y
-      LedgersFailure x -> Sum (LedgersFailure @era) 2 !> To x
-      TooManyExUnits x y -> Sum TooManyExUnits 3 !> To x !> To y
-      BodyRefScriptsSizeTooBig x y -> Sum BodyRefScriptsSizeTooBig 4 !> To x !> To y
+  encCBOR pf = ifEncodingVersionAtLeast (natVersion @10) (tenOnwards pf) (beforeTen pf)
```

Comment:
Since this code won't be in the release for Cheng+1, shouldn't this condition be for natVersion @11 ? I'm thinking, starting with version 10, failures will be encoded in the old way (as is now in master), but when your code kicks in, they will be decoded with this new code - won't this cause a problem?